### ISSUE TITLE
REST API compatibility issue on bulk API response

### ISSUE THREAD

**Issue by stevejgordon** [elastic/elasticsearch#85397]
### Elasticsearch Version

8.1.0

### Installed Plugins

N/A

### Java Version

_bundled_

### OS Version

Windows 10

### Problem Description

The language clients implemented support to enable API compatibility version headers to allow `7.x` clients to continue to function against an `8.x` server. Several consumers for the .NET client have [reported](https://github.com/elastic/elasticsearch-net/issues/6154) that our client considers the response from an `8.x` bulk request to be invalid. The reason is that we expect the `_type` field to be present on each operation item in the response.

Despite sending the `Accept: application/vnd.elasticsearch+json; compatible-with=7` header, the response does not include `_type` in the item object.

Expected Response (includes _type):
```
{
   "took": 30,
   "errors": false,
   "items": [
      {
         "index": {
            "_index": "test",
            "_type": "_doc",
            "_id": "1",
            "_version": 1,
            "result": "created",
            "_shards": {
               "total": 2,
               "successful": 1,
               "failed": 0
            },
            "status": 201,
            "_seq_no" : 0,
            "_primary_term": 1
         }
      }
   ]
}
```

Actual Response:
```
{
   "took": 30,
   "errors": false,
   "items": [
      {
         "index": {
            "_index": "test",
            "_id": "1",
            "_version": 1,
            "result": "created",
            "_shards": {
               "total": 2,
               "successful": 1,
               "failed": 0
            },
            "status": 201,
            "_seq_no" : 0,
            "_primary_term": 1
         }
      }
   ]
}
```

### Steps to Reproduce

1. Send a bulk request with an index operation to an `8.x` cluster with the API, with the `Accept` header set to ask for a v7 compatible response.

```
POST https://localhost:9200/bulk-testing/_bulk?pretty=true HTTP/1.1
Host: localhost
Accept: application/vnd.elasticsearch+json; compatible-with=7
Authorization: Basic ABCDEF=
Accept-Encoding: gzip, deflate
Content-Type: application/vnd.elasticsearch+json; compatible-with=7
Content-Length: 68

{"index":{"_id":123}}
{"value":"the value"}
```

**Comment by arteam** [comment]
I believe I fixed this issue in #84873

**Comment by stevejgordon** [comment]
@arteam Indeed you did! Sorry I missed that in my search. I just tested with 8.2.0-SNAPSHOT and everything looks good now. Thanks!
